### PR TITLE
TST: raising ValueError when inserting one dataframe in another

### LINKED ISSUE #42403: BUG: DafaFrame.insert doesn't raise an exception when inserting another DataFrame
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas
df = pandas.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
df.insert(1, "newcol", df)
df
   col1 newcol  col2
0     1   col1     3
1     2   col2     4
```

#### Problem description

Dataframe should take `int, Series, or array-like` as a `value` and throw an exception in case DataFrame is passed.

#### Expected Output

Behavior in 1.2.5

```python
import pandas
df = pandas.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
df.insert(1, "newcol", df)
ValueError: Wrong number of items passed 2, placement implies 1
```

#### Output of ``pd.show_versions()``

<details>

pandas           : 1.3.0
numpy            : 1.20.3
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 21.1.2
setuptools       : 49.6.0.post20210108
Cython           : None
pytest           : 6.2.4
hypothesis       : None
sphinx           : 4.0.2
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : 7.24.1
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : 2021.05.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : 0.15.0
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.3
sqlalchemy       : 1.4.18
tables           : 3.6.1
tabulate         : None
xarray           : 0.18.2
xlrd             : 2.0.1
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by debnathshoham**
- [x] closes #42403
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: pandas/tests/frame/indexing/test_insert.py [hunk 1]
```diff
@@ -89,3 +89,10 @@ def test_insert_item_cache(self, using_array_manager):
         ser.values[0] = 99
 
         assert df.iloc[0, 0] == df[0][0]
+
+    def test_insert_frame(self):
```
Comment (phofl) [inline pandas/tests/frame/indexing/test_insert.py:93]: Your test is broken

**Review by phofl — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/indexing/test_insert.py (hunk 1 above)
Comment (debnathshoham) [reply pandas/tests/frame/indexing/test_insert.py:93]: Very confused. I am getting a seperate error message on my local.

Comment (phofl) [reply pandas/tests/frame/indexing/test_insert.py:93]: The ci error is correct, you will have to debug locally why it is not raised for you

Comment (debnathshoham) [reply pandas/tests/frame/indexing/test_insert.py:93]: changed the error msg, but this is also failing. Would need some help, how same error raises multiple error msgs

**Review by jreback — APPROVED** [review]
lgtm. @phofl over to you

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Thanks @debnathshoham

**Pull request merged**